assignments/array-methods.js

```diff
@@ -56,28 +56,113 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+
+//ES5 WAY
+
+    // for(let i = 0; i < runners.length; i++) {
+    //     function addName()
+    //     {
+    //         console.log(`${runners[i].first_name} ${runners[i].last_name}`);
+    //     }
+    //     addName();
+    // }
+
+//ES6
+runners.forEach( (run) => {
+fullName.push(`${run.first_name} ${run.last_name}`);
+})
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
+//ES5 WAY
+// for(let i = 0; i < runners.length; i++) {
+//   function runnerName() {
+//     allCaps.push(runners[i].first_name.toUpperCase());
+//   }
+//   runnerName();
+// }
+// console.log(allCaps); 
+
+//ES6 Way
+runners.map(function(runner) {
+    allCaps.push(runner.first_name.toUpperCase())
+})
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
-console.log(largeShirts);
+//ES5
+// for(let i = 0; i < runners.length; i++)
+// {
+//     function runnerListFilter() 
+//     {
+//         if(runners[i].shirt_size === 'L') 
+//         {
+//             largeShirts.push(runners[i].first_name +" "+ runners[i].last_name);
+//         }
+//     }
+//     runnerListFilter();
+// }
+// console.log(largeShirts);
+
+// ES6 WAY
+    runners.filter(function(runnerShirtFilter) {
+        if (runnerShirtFilter.shirt_size === 'L')
+        {
+            largeShirts.push(`${runnerShirtFilter.first_name} ${runnerShirtFilter.last_name}`)
```

Comment:
you can assign `runners.filter()` into a variable so you don't need to use `.push()`